File: lab_sheet_7/checkpoint_3/IsPrime.java
import java.util.Scanner;

public class IsPrime {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number to test for primality: ");
        int num = scanner.nextInt();

        if (isPrime(num))
            System.out.println("The number is prime.");
        else
            System.out.println("The number is not prime.");
    }

    public static boolean isPrime(int num) {
        if (num == 2 || num == 3) {
            return true;
        }

        if ((num <= 1) || (num % 2 == 0) || (num % 3 == 0)) {
            return false;
        }

        int maxFactor = (int) Math.floor(Math.sqrt(num));

        for (int i = 2; i <= (maxFactor / 6); i++) {
            int calc = (int) i * 6;

            if (num % (calc - 1) == 0) {
                return false;
            }
            
            if (num % (calc + 1) == 0) {
                return false;
            }
        }

        return true;
    }
}

File: lab_sheet_7/checkpoint_2/is_prime.py

import math

def is_prime(num):
    if num <= 1:
        return False

    maxFactor = math.floor(math.sqrt(num))

    for i in range(2, maxFactor + 1):
        if num % i == 0:
            return False

    return True

def main():
    num = int(input("Enter a number to test for primality: "))

    if is_prime(num):
        print("The number is prime")
    else:
        print("The number is not prime")

if __name__ == "__main__":
    main()
